File: day_04/part01.py
#!/usr/bin/env python3

# advent of code 2017: day 04, part 01

with open("input.txt") as fp:
    data = fp.read().splitlines()
    fp.close

res = 0
for line in data:
    s = line.split()
    res = res + 1 if len(s) == len(set(s)) else res

print(res)

File: day_03/part01.go
/* advent of code 2017: day 03, part 01 */

package main

import (
    "fmt"
)

func abs(x int) int {
    if x < 0 {
        return -x
    }
    return x
}

func main() {
    var i, y, x int

    input := 325489
    for i=1; i*i < input; i+=2 {}
    calc := i*i
    done := false
    for y=i-1; y>=0 && !done; y-- {
        for x=i-1; x>=0 && !done; x-- {
            if (calc == input) {
                done = true
            }
            calc--;
        }
    }
    cy, cx := (i-1)/2, (i-1)/2
    res := abs((y/i) - (cy/i)) + abs((x%i) - (cx%i))
    fmt.Println(res)
}
